### ISSUE TITLE
Used profile named not correct displayed on last line at output

### ISSUE THREAD

**Issue by SchmidtJonathan1** [ansible/ansible-lint#3535]
##### Summary

There is a wrong output at the last check line ( Failed after **min** profile: 6 failure(s), 0 warning(s) on 1 files ). I used the shared profile defined at the config file or with a parameter, but the output shows all the time profile min.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.17.0 using ansible-core:2.15.0 ruamel-yaml:0.17.31 ruamel-yaml-clib:0.2.7
```

Ubuntu 22.04.2 LTS (WSL)

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

ansible-lint --profile shared -p prob.yml
ansible-lint -c .ansible-lint -p prob.yml

``` yaml
---
# .ansible-lint
profile: shared
```

```yaml
---
- hosts: localhost
  tasks:
    - name: install package
      shell: |
        yum install -y sos
    - name: Does not handle any output or return codes
      ansible.builtin.command: cat {{ my_file | quote }} # <- Does not handle the command output.
```

```console (paste below)
                       Rule Violation Summary                        
 count tag                       profile  rule associated tags       
     1 command-instead-of-module basic    command-shell, idiom       
     1 name[play]                basic    idiom                      
     1 name[casing]              moderate idiom                      
     2 no-changed-when           shared   command-shell, idempotency 

Failed after min profile: 5 failure(s), 0 warning(s) on 1 files.
```

##### Desired Behavior

Displayed the right profile at the last output line.

**Comment by ajinkyau** [comment]
Related: #2735
